Add date to Zed.log

### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

Would be useful if the Zed.log could add a date stamp at the boundary of a new day when something gets written to it. There is currently no way to see what day an entry was added to the log.
An example of where this would be good would be to make sure that when going through the log it adds what day it was created (maybe to check it wasn't from before a given release version?).

### Describe the solution you'd like

Zed adds a date stamp to day boundaries.

### If applicable, add mockups / screenshots to help present your vision of the feature

```
22:44:24 [INFO] set status to connected 3
22:44:24 [INFO] set status on client 0: Connected { connection_id: ConnectionId(3) }
2021-07-31 
00:24:05 [ERROR] connection error: error reading rpc message from socket

Caused by:
    0: IO error: Connection reset by peer (os error 54)
    1: Connection reset by peer (os error 54)
00:24:05 [INFO] set status on client 0: ConnectionLost
```

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -29,6 +29,7 @@ use settings::{
     self, settings_file::SettingsFile, KeymapFileContent, Settings, SettingsFileContent,
     WorkingDirectory,
 };
+use simplelog::ConfigBuilder;
 use smol::process::Command;
 use std::fs::OpenOptions;
 use std::{env, ffi::OsStr, panic, path::PathBuf, sync::Arc, thread, time::Duration};
@@ -269,8 +270,12 @@ fn init_logger() {
             .append(true)
             .open(&*paths::LOG)
             .expect("could not open logfile");
-        simplelog::WriteLogger::init(level, simplelog::Config::default(), log_file)
-            .expect("could not initialize logger");
+
+        let config = ConfigBuilder::new()
+            .set_time_format_str("%Y-%m-%dT%T") //All timestamps are UTC
+            .build();
+
+        simplelog::WriteLogger::init(level, config, log_file).expect("could not initialize logger");
     }
 }
 

